src/interval.cpp

```diff
@@ -7,33 +7,31 @@
 
 using namespace std;
 
-Interval::Interval(uint32_t s, uint32_t e) : start(s) {
-    assert(e >= start); // not a real interval ;
-    assert (e - start < std::numeric_limits<uint32_t>::max() or
-            assert_msg("Interval [" << s << "," << e << ") was too long"));
-    length = e - start; // intervals need to be exclusive of end point so that empty strings can be represented
+Interval::Interval(uint32_t s, uint32_t e) : start(s), end(e) {
+    assert(this->end >= this->start);
+    assert(this->end - this->start <= std::numeric_limits<uint32_t>::max());
+
+    // intervals must be end exclusive so that empty strings can be represented
+    this->length = this->end - this->start;
 }
 
 uint32_t Interval::get_end() const {
-    return start + (uint32_t) length;
+    return start + length;
```

Comment:
Would it not make more sense to just have this function return the `end` property?

Reply:
Do you mean this: `uint32_t Interval::get_end() const { return this->end; }` ?

Reply:
yes

Reply:
I guess, given that we always want `end >= start` we could make `start` and `end` private and have getter/setter functions for them so we can make sure `end >= start` holds.